diff --git a/password_policy.module b/password_policy.module
index 6b8d956..3d3158d 100644
--- a/password_policy.module
+++ b/password_policy.module
@@ -218,7 +218,8 @@ function _password_policy_constraints_validate(&$form, FormStateInterface &$form
 
   // Process user context
   // TODO - Turn this into configuration.
-  $user_context_fields = ['mail', 'name', 'uid'];
+  $user_context_fields = ['mail', 'name', 'uid', 'field_first_name', 'field_last_name'];
+//  $user_context_fields = ['mail', 'name', 'uid'];
   $user_context_values = [];
   foreach ($user_context_fields as $user_context_field) {
     $user_context_values[$user_context_field] = $form_state->getValue($user_context_field);
@@ -226,6 +227,9 @@ function _password_policy_constraints_validate(&$form, FormStateInterface &$form
     if ($user_context_field == 'uid') {
       $user_context_values[$user_context_field] = \Drupal::routeMatch()->getRawParameter('user');
     }
+    if (strpos($user_context_field, 'field_') !== false) {
+      $user_context_values[$user_context_field] = $user_context_values[$user_context_field][0]['value'];
+    }
     // Check default value.
     if (empty($user_context_values[$user_context_field]) and !empty($form['account'][$user_context_field]['#default_value'])) {
       $user_context_values[$user_context_field] = $form['account'][$user_context_field]['#default_value'];
